tests/core/code-stream/test_code_stream.py

```diff
@@ -118,3 +142,46 @@ def test_right_number_of_bytes_invalidated_after_pushxx():
     assert code_stream.is_valid_opcode(3) is False
     assert code_stream.is_valid_opcode(4) is True
     assert code_stream.is_valid_opcode(5) is False
+
+
+ALL_SINGLE_BYTES = tuple(sorted(set(range(0, 256))))
+PUSH_CODES = set(bytearray(range(96, 128)))
+NON_PUSH_CODES = set(ALL_SINGLE_BYTES).difference(PUSH_CODES)
+
+
+def _mk_bytecode(opcodes, data):
+    index_tracker = itertools.count(0)
+    for opcode in opcodes:
+        opcode_as_byte = bytes(bytearray((opcode,)))
+        if opcode in NON_PUSH_CODES:
+            yield next(index_tracker), opcode_as_byte
+        else:
+            data_size = opcode - 95
+            push_data = data.draw(st.binary(min_size=data_size, max_size=data_size))
+            yield next(index_tracker), opcode_as_byte + push_data
+            tuple(take(data_size, index_tracker))
```

Comment:
`take` was lazy, hence the `tuple` wrap.  Make sure `drop` works without forcing evaluation.

Reply:
Yup, I did, because I wasn't sure if it was lazy either. (I kind of expected it to return a new iterator and leave the argument unnaffected). The only reason I half-expected it to work is that `drop` does really sound like a side-effect-causing method. and it is!...

```
In [6]: from eth_utils.toolz import drop
In [7]: from itertools import count

In [8]: c = count()

In [9]: d = drop(2, c)

In [10]: next(d)
Out[10]: 2
```